### ISSUE TITLE
Thread frozen by lib with console.log

### ISSUE THREAD

**Issue by sogetimaitral** [lovell/sharp#690]
Hello,

We are leveraging the library to do some basic image resize on an existing nodejs app, but we have an issue we can't explain.

I have a code snippet that reproduces the issue, though it's not systematic but random, it's more than 80% failing.

Here is the code snippet for a full app.js, taken from the documentation about metadata (except the file is written as jpg in a file)

```
"use strict";

var sharp=require('sharp');

var image = sharp('/tmp/t.jpg');
image
  .metadata()
  .then(metadata => {
    console.log('size before resizing');
    return image
      .resize(Math.round(metadata.width / 2))
      .jpeg()
      .toFile('/tmp/ts.jpg');
  })
  .then(data => { });
```

When commenting the console.log line, it's working fine. But when it's there, it's blocking nodejs almost every time. If I press CTRL-C, the rest of the code succeed.

Do you have any clue on what's happening? Have you heard about this kind of issue with sharp?

Thanks!

Regards,
Greg

**Comment by lovell** [comment]
Hello, do you see any errors when you add a `catch` handler to the Promise chain?
```javascript
  .then(data => { })
  .catch(console.log);
```

**Comment by sogetimaitral** [comment]
Hello,

When adding the catch, nothing more is shown. This is coherent with the fact that a CTRL-C makes the program work again.
Were you able to reproduce the issue with the snippet? The behavior is really strange, and we can't figure out why, even with such a small snippet.

**Comment by lovell** [comment]
I am unable to reproduce this on OS X, Node v6.9.4, sharp v0.17.1. With what combinations are you seeing this behaviour?

The mention of Ctrl-C sounds like some form of process forking or piping is occurring external to your sample code.

**Comment by sogetimaitral** [comment]
I am using the same versions. But to get them, I restarted my linux, and now I don't have the issue anymore... 

I don't know what happened, and that's what is worrying, but it may be something totally unrelated. The code is running on a VM with 2 cores, maybe something was messed up on it until I rebooted it. I'll keep testing forward. 

Thanks for the answer, and sorry for the inconvenience :)
